frontend/test/storybook/visual-regression/v-filter-tab.spec.ts

```diff
@@ -11,51 +11,82 @@ test.describe.configure({ mode: "parallel" })
 const wrapper = "[role='tablist']"
 
 const focusFiltersTab = async (page: Page) => {
+  await expect(page.locator("#tab-filters")).toBeVisible()
   await page.keyboard.press("Tab")
+  const focusedTab = await page.evaluate(
+    () => document.activeElement?.textContent ?? ""
+  )
+  if (focusedTab.includes("Tab1")) {
+    await page.keyboard.press("ArrowRight")
+  }
 }
 
 const hoverFiltersTab = async (page: Page) => {
   await page.locator("#tab-filters").hover()
 }
+
+/**
+ * Fonts are the last request done by the storybook iframe. This function
+ * allows us to wait for the story to be fully-loaded before taking a screenshot.
+ */
+const goAndWaitForSvg = async (
+  page: Page,
+  args: Record<string, string | number | boolean>
+) => {
+  if (args.appliedFilterCount !== 0) {
+    await gotoWithArgs(page, args)
+  } else {
+    const response = page.waitForResponse(/\.svg/)
+    await gotoWithArgs(page, args)
+    await response
+    await expect(page.locator("#tab-filters svg")).toBeVisible()
+  }
+}
+
 test.describe("VFilterTab", () => {
   for (const appliedFilterCount of [0, 1, 12]) {
     /**
      * If the tab is focused, it will always be selected (it's not manually activated)
      */
-    test(`focused,${appliedFilterCount} filters`, async ({ page }) => {
-      await gotoWithArgs(page, { appliedFilterCount, isSelected: true })
+    test(`focused, ${appliedFilterCount} filters`, async ({ page }) => {
+      await goAndWaitForSvg(page, { appliedFilterCount, isSelected: true })
       await focusFiltersTab(page)
 
-      expect(await page.locator(wrapper).screenshot()).toMatchSnapshot(
+      await expect(page.locator(wrapper)).toHaveScreenshot(
```

Comment:
`await expect(...).toHaveScreenshot()`, nice!! That makes _so much sense_, to be able to wait for the screenshot as we expect it.

Does this also auto-update screenshots with `-u`, the way `toMatchSnapshot` does?

Reply:
I thought so too when I realized that it retries the screenshots. However, when I tried it, I saw that it makes the tests run much longer (~10 seconds instead of ~2 seconds), and, more importantly, doesn't always fix the problem. The icon-only search type button was still failing.